sql/updates/world/world_2018_11_24_01.sql

```diff
@@ -0,0 +1,3348 @@
+	ADD COLUMN `patch` TINYINT(3) UNSIGNED NOT NULL DEFAULT '0' AFTER `entry`,
+	DROP PRIMARY KEY,
+	ADD PRIMARY KEY (`entry`, `patch`);
+
+ALTER TABLE `item_enchantment_template`
+	ADD COLUMN `patch_min` TINYINT(3) UNSIGNED NOT NULL DEFAULT '0' AFTER `chance`,
+	ADD COLUMN `patch_max` TINYINT(3) UNSIGNED NOT NULL DEFAULT '21' AFTER `patch_min`;
+
+ALTER TABLE `battleground_template`
+	ADD COLUMN `patch` TINYINT(3) UNSIGNED NOT NULL DEFAULT '0' AFTER `id`,
+	DROP PRIMARY KEY,
+	ADD PRIMARY KEY (`id`, `patch`);
+
+ALTER TABLE `instance_template`
+	ADD COLUMN `patch` TINYINT(3) UNSIGNED NOT NULL DEFAULT '0' AFTER `map`,
+	DROP PRIMARY KEY,
+	ADD PRIMARY KEY (`map`, `patch`);
+
+ALTER TABLE `creature_loot_template`
+	ADD COLUMN `patch_min` TINYINT(3) UNSIGNED NOT NULL DEFAULT '0' AFTER `Comment`,
+	ADD COLUMN `patch_max` TINYINT(3) UNSIGNED NOT NULL DEFAULT '21' AFTER `patch_min`;
+
+ALTER TABLE `fishing_loot_template`
+	ADD COLUMN `patch_min` TINYINT(3) UNSIGNED NOT NULL DEFAULT '0' AFTER `Comment`,
+	ADD COLUMN `patch_max` TINYINT(3) UNSIGNED NOT NULL DEFAULT '21' AFTER `patch_min`;
+
+ALTER TABLE `gameobject_loot_template`
+	ADD COLUMN `patch_min` TINYINT(3) UNSIGNED NOT NULL DEFAULT '0' AFTER `Comment`,
+	ADD COLUMN `patch_max` TINYINT(3) UNSIGNED NOT NULL DEFAULT '21' AFTER `patch_min`;
+
+ALTER TABLE `item_loot_template`
+	ADD COLUMN `patch_min` TINYINT(3) UNSIGNED NOT NULL DEFAULT '0' AFTER `Comment`,
+	ADD COLUMN `patch_max` TINYINT(3) UNSIGNED NOT NULL DEFAULT '21' AFTER `patch_min`;
+
+ALTER TABLE `pickpocketing_loot_template`
+	ADD COLUMN `patch_min` TINYINT(3) UNSIGNED NOT NULL DEFAULT '0' AFTER `Comment`,
+	ADD COLUMN `patch_max` TINYINT(3) UNSIGNED NOT NULL DEFAULT '21' AFTER `patch_min`;
+
+ALTER TABLE `skinning_loot_template`
+	ADD COLUMN `patch_min` TINYINT(3) UNSIGNED NOT NULL DEFAULT '0' AFTER `Comment`,
+	ADD COLUMN `patch_max` TINYINT(3) UNSIGNED NOT NULL DEFAULT '21' AFTER `patch_min`;
+
+ALTER TABLE `disenchant_loot_template`
+	ADD COLUMN `patch_min` TINYINT(3) UNSIGNED NOT NULL DEFAULT '0' AFTER `Comment`,
+	ADD COLUMN `patch_max` TINYINT(3) UNSIGNED NOT NULL DEFAULT '21' AFTER `patch_min`;
+
+ALTER TABLE `prospecting_loot_template`
+	ADD COLUMN `patch_min` TINYINT(3) UNSIGNED NOT NULL DEFAULT '0' AFTER `Comment`,
+	ADD COLUMN `patch_max` TINYINT(3) UNSIGNED NOT NULL DEFAULT '21' AFTER `patch_min`;
+
+ALTER TABLE `reference_loot_template`
+	ADD COLUMN `patch_min` TINYINT(3) UNSIGNED NOT NULL DEFAULT '0' AFTER `Comment`,
+	ADD COLUMN `patch_max` TINYINT(3) UNSIGNED NOT NULL DEFAULT '21' AFTER `patch_min`;
+
+ALTER TABLE `mail_loot_template`
+	ADD COLUMN `patch_min` TINYINT(3) UNSIGNED NOT NULL DEFAULT '0' AFTER `Comment`,
+	ADD COLUMN `patch_max` TINYINT(3) UNSIGNED NOT NULL DEFAULT '21' AFTER `patch_min`;
+
+CREATE TABLE IF NOT EXISTS `forbidden_items` (
+  `entry` mediumint(8) unsigned NOT NULL DEFAULT 0,
+  `patch` tinyint(3) unsigned NOT NULL DEFAULT 0,
+  `AfterOrBefore` tinyint(3) unsigned NOT NULL DEFAULT 0,
```

Comment:
Not a very intuitive name. `after` instead?

Reply:
Also consider adding a foreign key from from forbidden_items to item_template (as well as change the engine type to do that), while we're in this new foreign key thing.

Reply:
I couldn't come up with a very good name for this column, but all it does is decide whether the item is forbidden after or before that patch (inclusive). It can be removed by making the `patch` column signed. Negative value for forbidden before, positive for forbidden after, but that would make the loading query uglier.

And if you are adding `patch_min` and `patch_max` columns to the vendor and loot tables, you could remove it entirely. I added it initially cause i wanted to avoid adding a patch column to loot tables, but then realized i have to because an item drop can be removed from a specific creature, but still drop from others. So this table is basically redundant if you take the time to assign patch to individual loot drops from the beginning.

Reply:
Thanks for the input. Seems resolved then, Kittnz added patch columns to loot table and removed forbidden_items.